src/main/java/org/truffleruby/language/LazyRubyNode.java

```diff
@@ -63,11 +63,13 @@ public RubyNode resolve() {
                 Log.LOGGER.info(() -> "lazy translating " + RubyLanguage.fileLine(getParent().getEncapsulatingSourceSection()) + " in " + getRootNode());
             }
 
-            resolved = resolver.get();
-            transferFlagsTo(resolved);
+            final RubyNode result = resolver.get();
+            transferFlagsTo(result);
 
-            replace(resolved, "lazy node resolved");
-            return resolved;
+            // publish
```

Comment:
What does this comment mean? That you need to publish the value at exactly this point? Why? If you fixed something and this comments is there to remind you, can you explain it further in the comment?